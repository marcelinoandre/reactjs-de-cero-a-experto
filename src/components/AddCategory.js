import { useState } from 'react';

const AddCategory = ({ setCategores }) => {
  const [category, setCategory] = useState('');

  const handleAdd = e => {
    e.preventDefault();
    if (!category) return console.log('vazio');
    setCategores(categores => [...categores, category]);
    setCategory('');
  };

  return (
    <form onSubmit={handleAdd}>
      <input
        name="category"
        type="text"
        value={category}
        onChange={e => setCategory(e.target.value)}
      />
      <button>Add</button>
    </form>
  );
};

export default AddCategory;
