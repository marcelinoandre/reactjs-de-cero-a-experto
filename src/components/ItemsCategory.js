import { useEffect, useState } from 'react';
import GifGridItem from '../GifGridItem';

const ItemsCategory = ({ categories }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url = `https://api.giphy.com/v1/gifs/search?q=rick+and+morty&limit=10&api_key=7Ktw4O8WmoytakpzYiaJYKaUIX32rQOj`;
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map(img => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url
      };
    });
    setImages(gifs);
  };

  return (
    <>
      {images.map(img => (
        <GifGridItem key={img.id} img={img} />
      ))}
    </>
  );
};

export default ItemsCategory;
