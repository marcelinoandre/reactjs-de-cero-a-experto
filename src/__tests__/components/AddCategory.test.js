import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('AddCategory', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  let submitForm = wrapper
    .find('form')
    .simulate('submit', { preventDefault() {} });

  beforeEach(() => {
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  it('should show corretly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should exchange the input text', () => {
    let input = wrapper.find('input');
    const valueTxt = 'Ola mundo';

    input.simulate('change', { target: { value: valueTxt } });

    expect(wrapper.find('p').text().trim()).toBe(valueTxt);
  });

  it('should return nothing if input is  empty  ', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  it('should  called the setCategories and clear the input text', () => {
    const cxText = wrapper.find('input');
    const category = 'Ola mundo';

    cxText.simulate('change', { target: { value: category } });

    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(cxText.text().trim()).toBe('');
  });
});
