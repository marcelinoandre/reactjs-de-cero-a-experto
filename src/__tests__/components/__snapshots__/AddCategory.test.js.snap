// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory should show corretly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
